packages/js/components/src/experimental-select-control/stories/index.tsx

```diff
@@ -133,36 +137,50 @@ export const FuzzyMatching: React.FC = () => {
 
 export const Async: React.FC = () => {
 	const [ selectedItem, setSelectedItem ] =
-		useState< SelectedType< DefaultItemType > >( null );
+		useState< DefaultItemType | null >( null );
 	const [ fetchedItems, setFetchedItems ] = useState< DefaultItemType[] >(
 		[]
 	);
 	const [ isFetching, setIsFetching ] = useState( false );
 
-	const fetchItems = ( value: string | undefined ) => {
-		setIsFetching( true );
-		setFetchedItems( [] );
-		setTimeout( () => {
-			const results = sampleItems.sort( () => 0.5 - Math.random() );
-			setFetchedItems( results );
+	const filter = useCallback(
+		( value = '' ) =>
+			new Promise< DefaultItemType[] >( ( resolve ) => {
+				setTimeout( () => {
+					const filteredItems = [ ...sampleItems ]
+						.sort( ( a, b ) => a.label.localeCompare( b.label ) )
+						.filter( ( { label } ) => label.includes( value ) );
+					resolve( filteredItems );
+				}, 1500 );
+			} ),
+		[ selectedItem ]
+	);
+
+	const selectProps = useAsyncFilter< DefaultItemType >( {
+		label: 'Async',
+		items: fetchedItems,
+		selected: selectedItem,
+		placeholder: 'Start typing...',
+		getFilteredItems: ( allItems ) => allItems,
+		onSelect: ( item: DefaultItemType | null ) => setSelectedItem( item ),
+		onRemove: () => setSelectedItem( null ),
+		filter,
+		onFilterStart() {
```

Comment:
> If we go with the above recommendation, I'd love to have a simple example in addition to this that doesn't need the onFilter... events, but this is a good example to show off its capability.

Do you mean to a different example using only the `filter` function, ignoring the others because they are optional right?